File: VB/SummarySample/CustomersViewModel.vb
Imports DevExpress.Mvvm
Imports System.Collections.Generic
Imports System.Collections.ObjectModel

Namespace SummarySample
	Public Enum SummaryItemType
		Max
		Count
		None
	End Enum

	Public Class CustomersViewModel
		Inherits ViewModelBase

		Public Sub New()
			Customers = CustomersDataModel.GetCustomers()
			TotalSummary = New ObservableCollection(Of Summary)() From {
				New Summary(SummaryItemType.Count, NameOf(Customer.Name)),
				New Summary(SummaryItemType.Max, NameOf(Customer.Visits))
			}
			GroupSummary = New ObservableCollection(Of Summary)() From {New Summary(SummaryItemType.Count, NameOf(Customer.Name))}
		End Sub
		Public ReadOnly Property Customers() As IList(Of Customer)
		Public ReadOnly Property TotalSummary() As ObservableCollection(Of Summary)
		Public ReadOnly Property GroupSummary() As ObservableCollection(Of Summary)
	End Class

	Public Class Summary
		Inherits BindableBase

		Public Sub New(ByVal type As SummaryItemType, ByVal fieldname As String)
			Me.Type = type
			Me.FieldName = fieldname
		End Sub
		Public ReadOnly Property Type() As SummaryItemType
		Public ReadOnly Property FieldName() As String
	End Class
End Namespace

File: CS/SummarySample/CustomersViewModel.cs
using DevExpress.Mvvm;
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace SummarySample {
    public enum SummaryItemType { Max, Count, None }

    public class CustomersViewModel : ViewModelBase {
        public CustomersViewModel() {
            Customers = CustomersDataModel.GetCustomers();
            TotalSummary = new ObservableCollection<Summary>() {
                new Summary( SummaryItemType.Count, nameof(Customer.Name) ),
                new Summary(SummaryItemType.Max, nameof(Customer.Visits) )
            };
            GroupSummary = new ObservableCollection<Summary>() {
                new Summary( SummaryItemType.Count, nameof(Customer.Name) )
            };
        }
        public IList<Customer> Customers { get; }
        public ObservableCollection<Summary> TotalSummary { get; }
        public ObservableCollection<Summary> GroupSummary { get; }
    }

    public class Summary : BindableBase {
        public Summary(SummaryItemType type, string fieldname) {
            Type = type;
            FieldName = fieldname;
        }
        public SummaryItemType Type { get; }
        public string FieldName { get; }
    }
}